### ISSUE TITLE
XCM v3-ready environment

### ISSUE THREAD

**Issue by hbulgarini** [paritytech/trappist#41]
Create a new local version of polkadot with XCM v3 changes back ported into it.

XCM v3 PR in Polkadot:
https://github.com/paritytech/polkadot/pulls?q=is%3Apr+is%3Aopen+xcm+v3

Main PR:  https://github.com/paritytech/polkadot/pull/4097

Cumulus companion account: https://github.com/paritytech/cumulus/pull/697

@arturgontijo

**Comment by bernardoaraujor** [comment]
please correct me if I'm wrong:

the work here can be split into three pieces:
- a `polkadot` executable with XCMv3 enabled
- a `polkadot-collator` executable with XCMv3 enabled
- a `trappist-collator` executable with XCMv3 enabled

**Comment by arturgontijo** [comment]
Leaving my "working" branches here as references:

https://github.com/arturgontijo/polkadot/tree/trappist-xcm-v3
https://github.com/arturgontijo/cumulus/tree/trappist-xcm-v3
https://github.com/arturgontijo/trappist/tree/xcm-v3

All of them are using a "static" substrate repo:
https://github.com/arturgontijo/substrate

**Comment by bernardoaraujor** [comment]
I updated the forks from @arturgontijo to be compatible with `v0.9.28`.

But I had to fork again, which started to get a bit too complex to maintain.
So I asked permission to @dvdplm and created some branches for us on the upstream repos.

The following branches have been successfully tested for asset teleport and reserve-based transfers with XCMv3 and `v0.9.28`:
- https://github.com/paritytech/polkadot/tree/trappist-xcm-v3
- https://github.com/paritytech/cumulus/tree/trappist-xcm-v3
- https://github.com/paritytech/trappist/tree/xcm-v3
